text/0000-attribute-actions.md

```diff
@@ -0,0 +1,244 @@
+- Start Date: 2015-10-17
+- RFC PR: https://github.com/emberjs/rfcs/pull/100
+- Ember Issue: (leave this empty)
+
+# Summary
+
+Ember's `action` helper has two inconsistent forms. The goal of this RFC is to
+allow the deprecation or removal of the classic action helper by adding a new
+DOM event listener API designed for closure actions.
+
+The RFC suggests three phases of implementation:
+
+* In Ember 2.x (after this RFC is merged), attribute actions like
+  `<button onclick={{action 'save'}}>` are declared an
+  official API.
+  * There are some very small code changes, but largely this is a
+    messaging and documentation change.
+  * These "attribute events" should use `element.addEventListener(eventName, handler, false)`
+    to attach instead of `element['on'+eventName'] = handler` for attachment.
+  * Non-action functions should warn if they are passed to `onclick=`
+    APIs.
+  * Non-functions should not be permitted as a bound value for `onclick=`, for
```

Comment:
I've certainly used `online=xxx` as an attribute name in the past, specific use-case being to show a badge on avatars for users who are currently online.

If it's matching `/on[a-z]+/` then it still allows using hyphenated attributes like `only-blank=true` which would otherwise be disallowed.

Whitelisting a few of the `on*` names could make sense. From that list I could see `one`, `once`, `only`, `onto` and `online` as not too unlikely to be used, some of which I've certainly used in the past.